homeassistant/helpers/storage.py

```diff
@@ -187,7 +187,9 @@ def _write_data(self, path: str, data: Dict):
             os.makedirs(os.path.dirname(path))
 
         _LOGGER.debug('Writing data for %s', self.key)
-        json.save_json(path, data, self._private)
+        tmp_path = path + "__TEMP__"
+        json.save_json(tmp_path, data, self._private)
+        os.replace(tmp_path, path)
```

Comment:
I think that we should move this functionality to `save_json` so that all calls can benefit. 

I also think that we should consider cleaning up the temp path in case of an exception replacing the other file

Reply:
Fair points. I'll make it so.